File: abb/TesteABB.java
import java.util.Random;
public class TesteABB {
    public static void main(String[] args) {
        ABB abb = new ABB();
        Random random = new Random();
        for (int i = 1; i <= 12; i++) {
            int n = random.nextInt(100) + 1;
            System.out.print(n + " ");
            abb.insere(n);
        }
        System.out.println("\n" + abb);
        System.out.println("numero de nos: " + abb.contaNos());
        System.out.println("soma dos valores: " + abb.somaValores());
    }
}

File: abb/teste_abb.c
#include <time.h>
#include "abb.h"

int main () {
    t_abb abb;
    int i, n;
    srand(time(0));
    inicia_arvore (&abb);
    for (i=1; i<=12; i++) {
        n = rand() % 100 + 1; //gera valores de 1 a 100
        printf ("%d ", n);
        insere (n, &abb);
    }
    printf ("\n");
    print_em_ordem (&abb);

    printf ("\na arvore tem %d nos", conta_nos(&abb));
    printf ("\na soma dos valores e %d", soma_valores(&abb));
    return 0;
}